src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Eventing/ExceptionsEventSource.cs

```diff
@@ -148,7 +145,29 @@ internal sealed class ExceptionsEventSource : EventSource
             SetValue(ref data[NameIdentificationEvents.TokenDescPayloads.OuterToken], OuterToken);
             SetValue(ref data[NameIdentificationEvents.TokenDescPayloads.Name], namePinned);
 
-            WriteEventCore(TokenDescriptionEventId, data);
+            WriteEventCore(ExceptionEvents.EventIds.TokenDescription, data);
+
+            RestartFlushingEventTimer();
+        }
+
+        [Event(ExceptionEvents.EventIds.Flush)]
+        public void Flush()
+        {
+            WriteEvent(ExceptionEvents.EventIds.Flush);
+        }
+
+        [NonEvent]
+        private void FlushTimerTick(object? state)
+        {
+            Flush();
+        }
+
+        [NonEvent]
+        private void RestartFlushingEventTimer()
+        {
+            // This will reset the timer to fire after the specified time period. If the timer is already
+            // started, it will be reset. If it already finished, it will be started again.
+            _flushEventsTimer.Change(TimeSpan.FromMilliseconds(100), Timeout.InfiniteTimeSpan);
```

Comment:
Does the `100` hold special significance here?

Reply:
Yes, I'll change it to a constant.